exporter/sumologicexporter/compress.go

```diff
@@ -39,11 +46,18 @@ func compressGZIP(data io.Reader) (io.Reader, error) {
 	return bytes.NewReader(buf.Bytes()), nil
 }
 
+// compressGZIP returns deflate compressed data
 func compressDeflate(data io.Reader) (io.Reader, error) {
```

Comment:
A bit more verbose? e.g.

```suggestion
// compressDeflate takes a reader with uncompressed data and returns
// a reader with the same data compressed using deflate algorithm
func compressDeflate(data io.Reader) (io.Reader, error) {
```